frappe/public/js/frappe/views/reports/report_view.js

```diff
@@ -509,7 +509,20 @@ frappe.views.ReportView = class ReportView extends frappe.views.ListView {
 					.then((updated_doc) => {
 						const _data = this.data.find(d => d.name === updated_doc.name);
 						for (let field in _data) {
-							_data[field] = updated_doc[field];
```

Comment:
It's fine to update each matching field with the value from `updated_doc`

The updated_doc object looks something like this `{<child table>: Array(2), modified_by: "Administrator", name: "82c6c0bacf", parent: null, creation: "2019-04-25 08:45:59.857965", …}`

So check the corresponding doctype of the `value` updated and see if it's a child table or not, if it's a childtable, then take the updated values from the childtable array of the updated doc object

Reply:
That's exactly what i'm doing, but for the entire row, because the data of the doc can be changed by hooks or other users interactions.

Reply:
the `if` statement that checks for `:` is therefore not necessary. It'll be safer to check the doctype property of `value`

Reply:
I think I'm not understanding what you want. Are you saying that I can replace this ```if(field.includes(':'))``` for something like this ```if(value.doctype == _data.doctype)```?? Do you want to update only the modified field??

I don't know if you need an explanation of what the code is doing, but here it is:
<details> 
  <summary>Spoiler </summary>

The ```for``` is looping all the fields of the edited row and updating their value with the value that is in the server after the edition is done correctly.

The ```if(field.includes(':'))``` is there because in ```_data``` the keys of the fields corresponding to the "parent" doctype have this format: "fieldname", and the fields of its child doctype have one like this: "child_doctype:fieldname".

So, before this patch, when the code said ```_data[field] = updated_doc[field]``` it was really doing ```_data["child_doctype:fieldname"] = updated_doc["child_doctype:fieldname"]```. In ```_data```, the key "child_doctype:fieldname" exists, but in ```updated_doc``` it does not.

Then, we check for the ":" in the key of ```_data```, if it's there it means that it's a child doctype's field and we have to split it and take the child doctype name (```cdt```) and fieldname (```_field```) separately.

Now we are ready to check every key of ```updated_doc``` to see wich of them is an array and if that array corresponds to the child doctype that we have in ```cdt```, if it does, then we take the one that is the correct for the row that was edited checking against the key of ```_data```: "child_doctype:name".

If after all this filters we get the correct one (```cdt_row```) we do ```_data[field] = cdt_row[_field]``` that is equal to ```_data["fieldname"] = updated_doc["child_table_field"]["correct_row_index"]["fieldname"]```.
</details>

Reply:
@scmmishra